pkg/generic/thrift/write.go

```diff
@@ -390,18 +390,28 @@ func writeBinaryList(ctx context.Context, val interface{}, out thrift.TProtocol,
 
 func writeList(ctx context.Context, val interface{}, out thrift.TProtocol, t *descriptor.TypeDescriptor, opt *writerOption) error {
 	l := val.([]interface{})
-	length := len(l)
+	length := 0
+	var sample interface{}
+	for _, elem := range l {
+		if elem != nil {
+			length++
+			sample = elem
```

Comment:
My idea is that if value is nil, don't just leave it out, but use the default value instead